### ISSUE TITLE
UnitTests: TestInitialize is not called before the test functions

### ISSUE THREAD

**Issue by FrankHentzschel** [rubberduck-vba/Rubberduck#3076]
The Function 
```
'@TestInitialize
Public Sub TestInitialize()
...
End Sub
```
I see the contents of data are not set as in TestInitialize, because of that I think it is not called correctly.

**Comment by shadowofsilicon** [comment]
I can confirm that there is some odd behavior with how the *Initialize and *Cleanup functions are called before/after the unit tests, as I have seen this while trying to work around the issue originally described in  #3074.  

I have three test modules, and I added Debug.Print messages to each ModuleInitialize, TestInitialize, TestCleanup, and ModuleCleanup function in each test module.

Originally, the unit tests would fail to run resulting in a message in the Test Explorer of:
'Test raised an error. Sorry, we couldn't find . Is it possible it was moved, renamed, or deleted?'
just like I wrote at the beginning of #3074.  But now, I see that the *Initialize and *Cleanup functions do not execute.

Suspecting as such, I added the object assignment lines from the ModuleInitialize to the unit test itself:
```
'@TestMethod
Public Sub TestMethod1() 'TODO Rename test
    On Error GoTo TestFail
    
    'Arrange:
    Set Assert = New Rubberduck.PermissiveAssertClass
    Set Fakes = New Rubberduck.FakesProvider
```
When I did this to 1 unit test in a module with 2 unit tests, both tests ran successfully and I could see debug messages as I would expect:
...
ModuleInitialize() in TestModule1 completed
TestInitialize() in TestModule1 completed
TestCleanup() in TestModule1 completed
TestInitialize() in TestModule1 completed
TestCleanup() in TestModule1 completed
ModuleCleanup() in TestModule1 completed
...

After a bunch of experimenting, it appears to me that once the objects are initialized via a unit test, Rubberduck successfully runs the *Initialize and *Cleanup functions, and will continue to successfully run them on repeated Run All button clicks, even if the object assignments that were added to the Unit Test has been commented out.  Additionally, it appears that once Rubberduck successfully runs the functions for one module, it can successfully run them for all the test modules that it has yet to iterate through.

Note, this is all except the one module with an underscore in the module name, I believe that is a unique bug which issue #3074 has become.

Note: this behavior is with a current build of the Next branch.

**Comment by shadowofsilicon** [comment]
FWIW, the debug log error message:

```
2017-06-19 14:51:49.8331;ERROR-2.0.14.16836;Rubberduck.UnitTesting.TestEngine;Unexpected COM exception while running tests.;System.Runtime.InteropServices.COMException (0x800A03EC): Sorry, we couldn't find . Is it possible it was moved, renamed or deleted?
   at System.Dynamic.ComRuntimeHelpers.CheckThrowException(Int32 hresult, ExcepInfo& excepInfo, UInt32 argErr, String message)
   at CallSite.Target(Closure , CallSite , ComObject , Object )
   at Rubberduck.VBEditor.Application.ExcelApp.Run(Object declaration) in C:\Users\electassy\Documents\Rubberduck\Rubberduck.VBEEditor\Application\ExcelApp.cs:line 17
   at Rubberduck.UnitTesting.TestEngine.Run(IEnumerable`1 members) in C:\Users\electassy\Documents\Rubberduck\RetailCoder.VBE\UnitTesting\TestEngine.cs:line 130
```

**Comment by bclothier** [comment]
Closing this issue as fixed because of PR #3778 -- verified that multiple methods with same names are no longer a problem for the `ExecuteCode` now and are executed in expected sequence.
